google/cloud/internal/populate_common_options.cc

```diff
@@ -35,6 +35,7 @@ Options PopulateCommonOptions(Options opts, std::string const& endpoint_env_var,
   if (!opts.has<EndpointOption>()) {
     auto e = internal::GetEnv(endpoint_env_var.c_str());
     if (e && !e->empty()) {
+      opts.set<AuthorityOption>(*e);
```

Comment:
(This isn't the sort of feedback you were looking for, but) do you want to override an already set `AuthorityOption`? We do not override an already set `EndpointOption`.

Reply:
> do you want to override an already set `AuthorityOption`?

I think so, but I'm open to be corrected.

> We do not override an already set `EndpointOption`.

Yeah, and that sounds like a mistake to me.  The environment variable should always win over anything set in the code.